microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoBroker.java

```diff
@@ -0,0 +1,648 @@
+        }
+    }
+
+    protected boolean handleReceivedMessage(final Socket receiveSocket, final MdpMessage msg) {
+        switch (msg.protocol) {
+        case C_CLIENT:
+            // Set reply return address to client sender
+            final Client client = clients.computeIfAbsent(msg.senderName, s -> new Client(receiveSocket, msg.protocol, msg.senderName, msg.senderID));
+            client.offerToQueue((MdpClientMessage) msg);
+            return true;
+        case W_WORKER:
+            processWorker(receiveSocket, (MdpWorkerMessage) msg);
+            return true;
+        default:
+            // N.B. not too verbose logging since we do not want that sloppy clients can bring down the broker through warning or info messages
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.atDebug().addArgument(msg).log("Majordomo broker invalid message: '{}'");
+            }
+            return false;
+        }
+    }
+
+    /**
+     * Process a request coming from a client.
+     */
+    protected void processClients() {
+        // round-robbin
+        clients.forEach((name, client) -> {
+            final MdpClientMessage clientMessage = client.pop();
+            if (clientMessage == null) {
+                return;
+            }
+            // dispatch client message to worker queue
+            final Service service = services.get(clientMessage.serviceName);
+            if (service == null) {
+                // not implemented -- according to Majordomo Management Interface (MMI) as defined in http://rfc.zeromq.org/spec:8
+                MajordomoProtocol.sendClientMessage(client.socket, MdpClientCommand.C_UNKNOWN, clientMessage.senderID, clientMessage.serviceNameBytes, "501".getBytes(StandardCharsets.UTF_8));
+                return;
+            }
+            // queue new client message RBAC-priority-based
+            service.putPrioritisedMessage(clientMessage);
+
+            // dispatch service
+            if (service.isInternal) {
+                final MdpClientMessage msg = service.getNextPrioritisedMessage();
+                if (msg == null) {
+                    // should be thrown only with VM '-ea' enabled -- assert noisily since this a (rare|design) library error
+                    assert false : "getNextPrioritisedMessage should not be null";
+                    return;
+                }
+                MajordomoProtocol.sendWorkerMessage(service.internalDispatchSocket, MdpWorkerCommand.W_REQUEST, null, msg.senderID, msg.payload);
+            } else {
+                //dispatch(requireService(clientMessage.serviceName, clientMessage.serviceNameBytes));
+                dispatch(service);
+            }
+        });
+    }
+
+    /**
+     * Process message sent to us by a worker.
+     */
+    protected void processWorker(final Socket receiveSocket, final MdpWorkerMessage msg) {
+        boolean workerReady = receiveSocket == internalServiceSocket || workers.containsKey(msg.senderIdHex);
```

Comment:
Codacy found an issue: [Use equals() to compare object references.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)